src/silentpayments.rs

```diff
@@ -0,0 +1,510 @@
+//! This module implements high-level Rust bindings for silent payments
+
+use core::ffi::c_void;
+use core::fmt;
+
+#[cfg(feature = "std")]
+use std;
+
+use core;
+
+use secp256k1_sys::{secp256k1_silentpayments_recipient_create_output_pubkey, secp256k1_silentpayments_recipient_create_shared_secret, secp256k1_silentpayments_recipient_public_data_create, secp256k1_silentpayments_recipient_public_data_parse, secp256k1_silentpayments_recipient_public_data_serialize, secp256k1_silentpayments_recipient_scan_outputs, secp256k1_silentpayments_sender_create_outputs, SilentpaymentsLabelLookupFunction};
+
+use crate::ffi::{self, CPtr};
+use crate::{constants, Keypair, PublicKey, SecretKey, XOnlyPublicKey};
+use crate::Secp256k1;
+use crate::Verification;
+
+fn copy_to_ffi_pubkey(pubkey: &PublicKey) -> ffi::PublicKey {
+
+    unsafe {
+        // Get a pointer to the inner ffi::PublicKey
+        let ffi_pubkey_ptr: *const ffi::PublicKey = pubkey.as_c_ptr();
+        
+        // Dereference the pointer to get the ffi::PublicKey
+        // Then create a copy of it
+        (*ffi_pubkey_ptr).clone()
+    }
+}
+
+
+/// Struct to store recipient data
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
```

Comment:
I think you can add `#[repr(transparent)]` here to avoid creating `ffi_recipients` since then the layout will be the same and you can soundly cast the pointers.

Reply:
Good catch! Done.